pkg/clustermesh/remote_cluster.go

```diff
@@ -129,3 +137,22 @@ func (rc *remoteCluster) Status() *models.RemoteCluster {
 	status.NumIdentities = int64(rc.remoteIdentityCache.NumEntries())
 	return status
 }
+
+func (rc *remoteCluster) onUpdateConfig(newConfig *cmtypes.CiliumClusterConfig) error {
+	oldConfig := rc.config
+
+	if newConfig != nil && oldConfig != nil && newConfig.ID == oldConfig.ID {
+		return nil
+	}
+	if newConfig != nil {
+		if err := rc.usedIds.reserveClusterID(newConfig.ID); err != nil {
+			return err
+		}
+		rc.config = newConfig
```

Comment:
This assignment should be moved to the end of the function, as it needs to be executed also in case the new config is nil. Otherwise, we might end up attempting to release the same cluster ID a second time, although it could have been acquired by another cluster.